src/IR2Vec.cpp

```diff
@@ -39,6 +39,9 @@ cl::opt<std::string> cl_iname(cl::Positional, cl::desc("Input file path"),
 
 cl::opt<std::string> cl_oname("o", cl::Required, cl::desc("Output file path"),
                               cl::cat(category));
+// for on demand generation of embeddings taking function name
+cl::opt<std::string> cl_fname("fname", cl::Optional, cl::init(""),
```

Comment:
fname could be perceived as file name as well. Please consider renaming it appropriately.